snap/snapcraft.yaml

```diff
@@ -835,3 +858,100 @@ parts:
+
+      cat "./cmd/res/configuration.toml" | \
+        sed -e s:\"./device-Modbus.log\":\'\$SNAP_COMMON/device-Modbus.log\': > \
+        "$SNAPCRAFT_PART_INSTALL/config/device-modbus/res/configuration.toml"
+
+      # FIXME: no Attribution.txt for 3rd party licenses; probably needs to be
+      # an aggregation of SDK's and this device service...
+
+      install -DT "./LICENSE-2.0.txt" \
+         "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-mongo/LICENSE-2.0.txt"
+  device-mqtt:
+    source: https://github.com/edgexfoundry/device-mqtt-go.git
+    source-depth: 1
+    #source-branch: dehli
+    plugin: make
+    after:
+      - go
+    override-build: |
+      . $SNAPCRAFT_STAGE/bin/go-build-helper.sh
+      gopartbootstrap github.com/edgexfoundry/device-mqtt-go
+
+      glide cc
+      make prepare
+      make build
+
+      install -DT "./cmd/device-mqtt" "$SNAPCRAFT_PART_INSTALL/bin/device-mqtt"
+
+      # FIXME: settings can't be overridden from the cmd-line!
+      # Override 'LogFile' and 'LoggingRemoteURL'
+      install -d "$SNAPCRAFT_PART_INSTALL/config/device-mqtt/res/"
+
+      cat "./cmd/res/configuration.toml" | \
+        sed -e s:\"./device-mqtt.log\":\'\$SNAP_COMMON/device-mqtt.log\': > \
+        "$SNAPCRAFT_PART_INSTALL/config/device-mqtt/res/configuration.toml"
+
+      install -T "./cmd/res/configuration-driver.toml" \
+        "$SNAPCRAFT_PART_INSTALL/config/device-mqtt/res/configuration-driver.toml"
+
+      # FIXME: no Attribution.txt for 3rd party licenses; probably needs to be
+      # an aggregation of SDK's and this device service...
+
+      install -DT "./LICENSE-2.0.txt" \
+         "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-mqtt/LICENSE-2.0.txt"
+  device-random:
+    source: https://github.com/edgexfoundry/device-random.git
+    source-depth: 1
+    #source-branch: dehli
+    plugin: make
+    after:
+      - go
+    override-build: |
+      . $SNAPCRAFT_STAGE/bin/go-build-helper.sh
+      gopartbootstrap github.com/edgexfoundry/device-random
+
+      glide cc
+      make prepare
+      make build
+
+      install -DT "./cmd/device-random" "$SNAPCRAFT_PART_INSTALL/bin/device-random"
+
+      # FIXME: settings can't be overridden from the cmd-line!
+      # Override 'LogFile'
+      install -d "$SNAPCRAFT_PART_INSTALL/config/device-random/res/"
+
+      # install configuratin & default device profile
+      cat "./cmd/res/configuration.toml" | \
+        sed -e s:\"./device-random.log\":\'\$SNAP_COMMON/device-random.log\': > \
+        "$SNAPCRAFT_PART_INSTALL/config/device-random/res/configuration.toml"
+
+      install -T "./cmd/res/device.random.yaml" \
+        "$SNAPCRAFT_PART_INSTALL/config/device-random/res/device.random.yaml"
+
+      # FIXME: no Attribution.txt for 3rd party licenses; probably needs to be
+      # an aggregation of SDK's and this device service...
+
+      install -DT "./LICENSE-2.0.txt" \
+         "$SNAPCRAFT_PART_INSTALL/usr/share/doc/device-random/LICENSE-2.0.txt"
```

Comment:
So close to 1000 line snapcraft.yaml!!! We should just add another 43 lines of comments or whitespace :tada: